### ISSUE TITLE
Allow single line closures

### ISSUE THREAD

**Issue by CarsonF** [squizlabs/PHP_CodeSniffer#1580]
This code:
```php
$func = function ($item) { return $item['foo']; };
```
Triggers:
- `Generic.Formatting.DisallowMultipleStatements.SameLine`
- `PEAR.WhiteSpace.ScopeClosingBrace.Line`
- `Squiz.WhiteSpace.ScopeClosingBrace.ContentBefore`
- `Squiz.Functions.MultiLineFunctionDeclaration.ContentAfterBrace`

---

IMO the `DisallowMultipleStatements` error is a false positive. The code above is not the same as
```php
$a = ''; $b = '';
```
----

For the rest of the errors, it would be nice to the have an option to allow single statement methods to be on one line (PHPStorm has this option). Maybe another option to further restrict it to only closures.

**Comment by tgr** [comment]
Referenced in https://phabricator.wikimedia.org/T154789

**Comment by gsherwood** [comment]
> would you entertain a PR for this?

I can't see how this could be done with the existing sniffs. This is a specific rule for closures that does not exist in any of the included standards and is probably best served using a custom sniff. Including a rule like "Allow single line closures if they only have one statement" in these sniffs is far too specific.

**Comment by gsherwood** [comment]
Closing due to lack of additional interest in a new sniff.
